### ISSUE TITLE
Pybel crashes with Anaconda Python 3.x

### ISSUE THREAD

**Issue by ghutchis** [openbabel/openbabel#1802]
- [X] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: All
Operating system and version: MacOS with Anaconda (probably not limited)

## Actual Behavior
```
% /usr/local/anaconda/bin/python3
> import openbabel
Fatal Python error: PyThreadState_Get: no current thread
Abort trap: 6
```
## Solution
Anaconda now performs static linking on python3, so there's no libpython.so or libpython.dylib. This means that _openbabel.so needs different linker flags:

On Mac:
`-bundle -undefined dynamic_lookup`

This can be retrieved using Python sysconfig:
```
>>> import sysconfig
>>> sysconfig.get_config_var("LDSHARED")
'gcc -bundle -undefined dynamic_lookup -L/usr/local/anaconda/lib  -arch x86_64 -L/usr/local/anaconda/lib  -arch x86_64'
```

**Comment by mwojcikowski** [comment]
Should I make a provisional patch in conda recipe to fix this issue for 2.4.1? I will try to reproduce it on linux, but I can only test it on Travis since I dont have a Mac at hand.

**Comment by ghutchis** [comment]
Seems like a good idea. I don't have time until this weekend, but I was assuming I'd add LDFLAGS into the CMakefile for Mac as well.
